### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by ecekpo**
**Updated changes**
- Declare callback method as private for post, like and comment models

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  after_create :post_counter_update
+
```
Comment (Ghiftee) [inline app/models/post.rb:7]: - [ ] It is considered good practice to declare callback methods as private. If left public, they can be called from outside of the model and violate the principle of object encapsulation. Kindly, make ```post_counter_update``` private.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_create :update_like_counter
+
+  def update_like_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (Ghiftee) [inline app/models/like.rb:9]: - [ ] It is considered good practice to declare callback methods as private. If left public, they can be called from outside of the model and violate the principle of object encapsulation. Kindly, make ```update_like_counter``` private.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_create :update_comment_counter
+
+  def update_comment_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (Ghiftee) [inline app/models/comment.rb:9]: - [ ] It is considered good practice to declare callback methods as private. If left public, they can be called from outside of the model and violate the principle of object encapsulation. Kindly, make ```update_comment_counter``` private.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @ecekpo  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ You created models
✔️ Associations are set according to the ERD diagram

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by fobadara — APPROVED** [review]
Hi @ecekpo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
✅ Project looks professional

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**